types/types.go

```diff
@@ -995,12 +995,18 @@ func GetLHVolumeAttachmentNameFromVolumeName(volName string) string {
 
 // IsSelectorsInTags checks if all the selectors are present in the tags slice.
 // It returns true if all selectors are found, false otherwise.
-func IsSelectorsInTags(tags, selectors []string) bool {
+func IsSelectorsInTags(tags, selectors []string, tolerateEmptySelector bool) bool {
 	if !sort.StringsAreSorted(tags) {
 		logrus.Debug("BUG: Tags are not sorted, sorting now")
 		sort.Strings(tags)
 	}
 
+	if len(selectors) == 0 {
+		if !tolerateEmptySelector && len(tags) != 0 {
+			return false
+		}
+	}
```

Comment:
because this condition may miss the following case
```
tags = ["A"], selector = [], emptySelectorAllowed=false
```
and since the selector is empy so it also skips the following iteration of selector list
then return true directly, which should be falise since tags = ["A"] is not empy

Reply:
when selectors is empty
- if not allow empty selector
    - if tags = [] => return true
    - if tags = ['A'] => return false

There are still two condition when not allow empty selector